sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileApiTests.java

```diff
@@ -2416,6 +2416,16 @@ public void renameMin() {
         assertNotNull(primaryFileClient.rename(generatePathName()));
     }
 
+    @RequiredServiceVersion(clazz = ShareServiceVersion.class, min = "2021-04-10")
+    @ParameterizedTest
+    @ValueSource(strings = {"\u200B", "\u200C", "\u200D", "\uFEFF"})
+    public void renameWithUnicodeChars(String specialChar) {
+        ShareFileClient fileClient = shareClient.getFileClient("test-file-source" + specialChar + " pdf.txt");
+        fileClient.create(512);
+        ShareFileClient destClient = fileClient.rename("test-file-destination" + specialChar + " pdf.txt");
+        assertNotNull(destClient);
```

Comment:
Is there any validation we want to do on the destination client name?

Reply:
Adding an assertion to make sure the special characters are included in the destination client.